ui/cypress/e2e/sampleControls.cy.js

```diff
@@ -0,0 +1,41 @@
+/* global cy, it, describe, beforeEach, Cypress */
+
+describe('3-click centring', () => {
+  beforeEach(() => {
+    cy.login();
+    cy.findByRole('link', { name: 'MXCuBE-Web (OSC)' }).should('be.visible');
+    cy.takeControl();
+  });
+
+  it('3-click centring should not work without sample', () => {
+    cy.clearSamples();
+    cy.contains('button', '3-click centring').click();
+    cy.findByRole('alert', 'Error: There is no sample mounted').should(
+      'be.visible',
+    );
+  });
+
+  it('Each click is rotating the sample by 90 degrees', () => {
+    cy.mountSample();
+    cy.contains('button', '3-click centring').click();
+    /* eslint-disable-next-line promise/catch-or-return */
+    cy.get('.form-control[name="diffractometer.phi"]')
+      .invoke('val')
+      /* eslint-disable-next-line promise/prefer-await-to-then */
+      .then((value) => {
+        let omegaValue = Number.parseFloat(value);
+        Cypress._.times(2, () => {
+          omegaValue += 90;
+          cy.get('.canvas-container').click();
+          // to update the omega value, a small amount of time must be waited and the page reloaded
+          cy.wait(1000);
+          cy.reload();
+          // press the centring button again after reload to stay in the correct mode
+          cy.contains('button', '3-click centring').click();
+          cy.get('.form-control[name="diffractometer.phi"]')
+            .invoke('val')
+            .should('equal', omegaValue.toFixed(2));
```

Comment:
Would be good to act more like a real user here (and in other places like in the `takeControl` command), with queries like [`cy.findByLabelText`](https://testing-library.com/docs/queries/bylabeltext) (to find an input field with its label) or [`cy.findByRole`](https://testing-library.com/docs/queries/byrole) (to find headings, buttons, etc.).

It can be done later though, as it may require "fixing" the accessibility/markup of the relevant features (like labeling form inputs).

Reply:
As you mentioned for some elements, I could not use the mentioned queries. However, I agree with you that I often did not act like a real user (especially for `takeControl` and `clearSamples` commands) and some parts can be handled differently, I will take a look at it and make the necessary changes. Thanks for the comment!

Reply:
:+1: Thanks guys !